apis/python/requirements-dev.txt

```diff
@@ -0,0 +1,7 @@
+black
```

Comment:
What is the difference between the dev requirements and requirements for the package? (This is genuine question I don't know :D )

Reply:
Similar to my comment above (https://github.com/TileDB-Inc/TileDB-VCF/pull/701/files#r1565992899), I agree that this file `requirements-dev.txt` is confusing because it includes more than just development-only dependencies. Here is how I would classify them based on my current understanding:

* dev (only needed for local development and CI): black
* runtime (required to `import tiledbvcf` because they are hit by `__init__.py`): pandas, pyarrow, pyarrow-hotfix
* optional (conditionally used if installed): dask
* test (required to run the test suite): dask, pytest, tiledb

Reply:
The `requirements-dev.txt` was only there for building a local dev environment. As mentioned above, we'll make the dependencies clear in a more standard way.